graylog2-web-interface/src/views/components/datatable/DataTable.tsx

```diff
@@ -128,6 +163,32 @@ const DataTable = ({
     return Promise.reject();
   }, [config, widget, editing, formContext]);
 
+  const togglePin = useCallback((field: string) => {
+    const dirty = formContext?.dirty;
+
+    const updateWidget = () => {
+      const curVisualizationConfig = widget.config.visualizationConfig || DataTableVisualizationConfig.create([]).toBuilder().build();
+      let pinned_columns;
+
+      if (curVisualizationConfig?.pinned_columns?.has(field)) {
```

Comment:
We could make this a bit more concise:
```suggestion
      const pinnedColumns = curVisualizationConfig?.pinned_columns?.has(field)
          ? curVisualizationConfig.pinned_columns.delete(field)
          : curVisualizationConfig.pinned_columns.add(field);
```